Failed to build with rustc 1.21.0-nightly (230a379a4 2017-08-18)

Due to: rust-lang/rust#43832

Logs:

```
error[E0308]: mismatched types
   --> .cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.152/src/utils/mod.rs:117:61
    |
117 |             ExpnFormat::CompilerDesugaring(name) => name != "...",
    |                                                             ^^^^^ expected enum `syntax_pos::CompilerDesugaringKind`, found reference
    |
    = note: expected type `syntax_pos::CompilerDesugaringKind`
               found type `&'static str`

error: aborting due to previous error

error: Could not compile `clippy_lints`.
warning: build failed, waiting for other jobs to finish...
error: failed to compile `clippy v0.0.152`, intermediate artifacts can be found at `/tmp/cargo-install.zAvvHQr48G8S`

Caused by:
  build failed
```

--- a/clippy_lints/src/utils/mod.rs
+++ b/clippy_lints/src/utils/mod.rs
@@ -16,7 +16,7 @@ use std::mem;
 use std::str::FromStr;
 use syntax::ast::{self, LitKind};
 use syntax::attr;
-use syntax::codemap::{ExpnFormat, ExpnInfo, Span, DUMMY_SP};
+use syntax::codemap::{CompilerDesugaringKind, ExpnFormat, ExpnInfo, Span, DUMMY_SP};
 use syntax::errors::DiagnosticBuilder;
 use syntax::ptr::P;
 use syntax::symbol::keywords;
@@ -114,7 +114,7 @@ pub fn in_constant(cx: &LateContext, id: NodeId) -> bool {
 pub fn in_macro(span: Span) -> bool {
     span.ctxt.outer().expn_info().map_or(false, |info| {
         match info.callee.format {// don't treat range expressions desugared to structs as "in_macro"
-            ExpnFormat::CompilerDesugaring(name) => name != "...",
+            ExpnFormat::CompilerDesugaring(kind) => kind != CompilerDesugaringKind::DotFill,
             _ => true,
         }
     })

